authentik/root/settings.py

```diff
@@ -305,14 +305,15 @@
     }
 }
 
-if CONFIG.get_bool("postgresql.use_pgpool", False):
-    DATABASES["default"]["DISABLE_SERVER_SIDE_CURSORS"] = True
-
-if CONFIG.get_bool("postgresql.use_pgbouncer", False):
-    # https://docs.djangoproject.com/en/4.0/ref/databases/#transaction-pooling-server-side-cursors
-    DATABASES["default"]["DISABLE_SERVER_SIDE_CURSORS"] = True
-    # https://docs.djangoproject.com/en/4.0/ref/databases/#persistent-connections
-    DATABASES["default"]["CONN_MAX_AGE"] = None  # persistent
+# https://docs.djangoproject.com/en/4.0/ref/databases/#transaction-pooling-server-side-cursors
+DATABASES["default"]["DISABLE_SERVER_SIDE_CURSORS"] = CONFIG.get_bool(
+    "postgresql.disable_server_side_cursors", False
+)
+# https://docs.djangoproject.com/en/4.0/ref/databases/#persistent-connections
+# Not a PostgreSQL-specific setting, but a database-specific setting.
+# Only PostgreSQL is supported as a database, so we place this setting it in the 'postgresql' path.
+DATABASES["default"]["CONN_MAX_AGE"] = CONFIG.get("postgresql.conn_max_age", None)
```

Comment:
```suggestion
DATABASES["default"]["CONN_MAX_AGE"] = CONFIG.get("postgresql.conn_max_age", 0)
```
Which is the default. Also, we should use `get_int` here, but we need to figure out a way for that to be able to return `None` if that's what is set by users.

Reply:
> Also, we should use `get_int` here, but we need to figure out a way for that to be able to return None if that's what is set by users.

I see a problem:
* Making `get_int` return `None` would require changing the type signature to `int | None` (regardless of how it's implemented). I _think_ that would add too much of a mental load as every call to `get_int` now needs to account for `None`.

For this reason, I would define a new `get_optional_int` with the `-> int | None` signature, and would check for env variables set to the string `None` (case-insensitive). This is similar to how `get_bool` works, and also makes it obvious this function may not return an int.

If you disagree, though, I can move the implementation to `get_int` instead, and hope there aren't many type errors to wrap around a `if not None else default`.

Reply:
On another topic: Agree with setting `0` as the default `CONN_MAX_AGE`, as that's the Django actual default, I've made this change too :+1:

Reply:
Yeah I agree about the `get_optional_int`. `CONN_MAX_AGE` still defaults to None from what I'm reading though

Reply:
> `CONN_MAX_AGE` still defaults to None from what I'm reading though

Only in the documentation, my bad